kong/plugins/prometheus/exporter.lua

```diff
@@ -336,6 +368,38 @@ local function metric_data()
     enterprise.metric_data()
   end
 
+  -- Hybrid mode status
+  if cp_metrics then
+    -- Cleanup old metrics
+    metrics.data_plane_last_seen:reset()
+    metrics.data_plane_config_hash:reset()
```

Comment:
The code will reset the metrics and then could fail to list data planes on line 377.
I suggest we do the following:
1. list the data planes , check err
2. reset metrics
3. start the loop to populate the metrics.

Reply:
I think it will be useful we just fail hard and not emitting any DP metrics at all on DB error.
Actually if we list DPs first, there'll still be possiblity that we are exposing partial data.
The one case that can be solved is that no DPs are listed at all, and then we keep exposing the old metrics.
but that may silencing the errors to end user, since if the plugin does fail to list metrics continously,
prometheus side its not detectable at all. Maybe a "kong_prometheus_errors" counter will help but
that's a different story.

Reply:
This highlights another problem. From what I can read, it seems every time there is a request on `/metrics`, there will be a SQL query. Is that right? If yes, that seems too much computation to do for each request on `/metrics`. Can we please introduce some form of caching here?

Reply:
Synced with @hbagdi offline, we will continue with current approach for now to avoid adding complexity in code.
Added https://github.com/Kong/kong-plugin-prometheus/issues/132 for tracking it.